invenio_preservation_sync/api.py

```diff
@@ -0,0 +1,163 @@
+        )
+
+        db.session.add(preservation)
+        db.session.commit()
+        return preservation
+
+    @classmethod
+    def get_by_record_id(cls, record_id):
+        """Get Preservation Info by record id."""
+        preservations = PreservationInfo.query.filter_by(record_id=record_id).order_by(
+            text(cls.query_order)
+        )
+        return preservations
+
+    @classmethod
+    def get_latest_by_record_id(cls, record_id):
+        """Get latest Preservation Info for the given record id."""
+        preservation = (
+            PreservationInfo.query.filter_by(record_id=record_id)
+            .order_by(text(cls.query_order))
+            .first()
+        )
+        return preservation
+
+    @classmethod
+    def get_existing_preservation(
+        cls,
+        record_id=None,
+        revision_id=None,
+        archive_timestamp=None,
+    ):
+        """Return Preservation Info if it already exists."""
+        preservation = PreservationInfo.query.filter_by(
+            record_id=record_id,
+            revision_id=revision_id,
+            archive_timestamp=archive_timestamp,
+        ).first()
+        return preservation
+
+    @classmethod
+    def update_existing_preservation(
+        cls,
+        preservation,
+        status=None,
+        harvest_timestamp=None,
+        uri=None,
+        path=None,
+        event=None,
+        description=None,
+    ):
+        """Update existing preservation."""
+        status = PreservationInfoAPI.convert_status(status)
+
+        if (
+            preservation.status == status
+            and preservation.harvest_timestamp == parse(harvest_timestamp)
+            and preservation.uri == uri
+            and preservation.path == path
+            and preservation.description == description
+        ):
+            raise PreservationAlreadyReceivedError(preservation)
+
+        if status:
+            preservation.status = status
+
+        if harvest_timestamp:
+            preservation.harvest_timestamp = harvest_timestamp
+
+        if uri:
+            preservation.uri = uri
+
+        if path:
+            preservation.path = path
+
+        if description:
+            preservation.description = description
+
+        preservation.event = event
+
+        db.session.commit()
+
+    @classmethod
+    def convert_status(cls, value):
+        """Convert the status of the preservation info.
+
+        Valid examples: "P", "Preserved", "preserved"
+        """
+        if isinstance(value, str):
+            if value in [x.value for x in PreservationStatus]:
+                return PreservationStatus(value)
+            elif value.upper() in [x.name for x in PreservationStatus]:
+                return PreservationStatus[value.upper()]
+        elif isinstance(value, PreservationStatus):
+            return value
+        raise ValueError(
+            f"Invalid status value for the Preservation Status. Got: {value}"
+        )
```

Comment:
```suggestion
    if isinstance(value, PreservationStatus):
        return value
    elif isinstance(value, str):
        return PreservationStatus(value.upper())
    else:
        raise TypeError("Value must be a PreservationStatus or a string.")
```

Usages of this:

```python

In [8]: check_status(PreservationStatus('F'))
Out[8]: <PreservationStatus.FAILED: 'F'>

In [9]: check_status('f')
Out[9]: <PreservationStatus.FAILED: 'F'>

In [10]: check_status('F')
Out[10]: <PreservationStatus.FAILED: 'F'>

In [11]: check_status('FF')
Out [11] ValueError: 'FF' is not a valid PreservationStatus

In [12]: check_status(PreservationStatus('FF'))
Out [12]: ValueError: 'FF' is not a valid PreservationStatus
```

Reply:
As discussed: used your suggestions but added back the possibility to use the `name` from to enum to add more flexibility